### PR TITLE
feat: add table

### PR DESCRIPTION
**Pull request by dominique-pfister**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,8 +10,8 @@
   "type": "module",
   "types": "src/index.d.ts",
   "scripts": {
-    "test": "c8 mocha",
-    "test-ci": "c8 mocha && codecov",
+    "test": "c8 mocha --spec=test/**/*.test.js",
+    "test-ci": "c8 mocha  --spec=test/**/*.test.js && codecov",
```
Comment (dominique-pfister) [inline package.json:14]: adding back spec here

File: src/excel/Workbook.d.ts [hunk 2]
```diff
@@ -42,6 +42,16 @@ export declare interface Workbook {
...
 
+  /**
+   * Add a table to a workbook. Note, that it is not possible
+   * to define the name at this point, one has to rename the
+   * table afterwards (see Table#rename).
+   *
+   * @param address address of table, e.g. Sheet1!A1:C1
+   * @param hasHeaders whether the table has headers
+   */
+  addTable(address, hasHeaders): Promise<Table>;
```
Comment (tripodsan) [inline src/excel/Workbook.d.ts:53]: what about adding an implicit `rename` after creation, if a name is given?
this should also check if the table with the given name already exists.

Comment (dominique-pfister) [reply src/excel/Workbook.d.ts:53]: Nice one 💡

File: src/excel/Workbook.js [hunk 3]
```diff
@@ -47,6 +48,27 @@ export class Workbook extends NamedItemContainer {
     return new Table(this._oneDrive, `${this._uri}/tables`, name, this._log);
   }
 
+  async addTable(address, hasHeaders, name) {
+    if (name) {
+      const names = await this.getTableNames();
+      if (names.includes(name)) {
+        throw new StatusCodeError(`Table name already exists: ${name}`, 409);
+      }
+    }
```
Comment (dominique-pfister) [inline src/excel/Workbook.js:57]: check whether name already exists, if given

File: src/excel/Workbook.js [hunk 4]
```diff
@@ -47,6 +48,27 @@ export class Workbook extends NamedItemContainer {
...
+      method: 'POST',
+      body: {
+        address,
+        hasHeaders,
+      },
+    });
+    const table = this.table(result.name);
+    if (name && name !== table.name) {
+      await table.rename(name);
+    }
```
Comment (dominique-pfister) [inline src/excel/Workbook.js:68]: if name is given and it doesn't equal the automatically generated one, do a `rename`

File: src/excel/Workbook.d.ts (hunk 2 above)
Comment (dominique-pfister) [reply src/excel/Workbook.d.ts]: Please re-verify

**Review by tripodsan — APPROVED** [review]

**Pull request merged**